include/pocl.h

```diff
@@ -506,6 +506,17 @@ typedef union
 typedef struct _pocl_buffer_migration_info
   pocl_buffer_migration_info;
 
+/**
+ * Enum that represents different possible states of a command node.
+ */
+typedef enum
+{
+  POCL_COMMAND_FAILED = -1,
```

Comment:
Can you add some examples when this state could be invoked. Maybe we can use it for implementing #1619 also?

Reply:
I'm currently using it in the remote driver to mark a command node as failed when the device is not available. The remote driver does some bookkeeping regarding inflight commands and I'm using this to handle things a bit differently when it fails.
It shares a lot of purpose with the status of the command node's sync event, but since it is possible for the node to use something else than events, I've resorted to this instead.

Reply:
regarding the issue you mentioned, it could work. I'd have to check the code to be sure, but I believe that an event failing will cascade that to the rest of the events. 

And as it so happened, I recently did an out of bounds read in a kernel on the Nvidia OpenCL library and it would not crash, but instead return a CL_OUT_OF_RESOURCES when trying to read the output buffer. It was a bit confusing to figure why it was happening, but it does communicate to the host something is wrong.